src/dynamics/video_player/player/states.js

```diff
@@ -471,41 +472,46 @@ Scoped.define("module:VideoPlayer.Dynamics.PlayerStates.LoadAds", [
         _started: function() {
             if (this.dyn.get("adshassource")) {
                 if (this._triggerLoadAds()) {
-                    if (!this.dyn.get("adsplayer_active")) this.dyn.set("adsplayer_active", true);
-                    this.dyn.channel("ads").trigger("load");
-                    this.listenOn(this.dyn.channel("ads"), "contentResumeRequested", function() {
-                        if (this.dyn.get("adtagurlfallbacks") && this.dyn.get("adtagurlfallbacks").length > 0) {
-                            this.dyn.set("adtagurl", this.dyn.get("adtagurlfallbacks").shift());
-                            this.dyn.scopes.adsplayer.execute("requestAds");
-                        }
-                        this.next(this._nextState());
-                    }, this);
-                    this.listenOn(this.dyn.channel("ads"), "loaded", function() {
-                        this.next(this._nextState());
-                    }, this);
-                    this.listenOn(this.dyn.channel("ads"), "log", function(event) {
-                        if (!event.getAdData().adError || !this.dyn.get("adtagurlfallbacks") || this.dyn.get("adtagurlfallbacks").length === 0) return;
-                        this.dyn.set("adtagurl", this.dyn.get("adtagurlfallbacks").shift());
-                        this.dyn.brakeAdsManually();
+                    if (this.dyn.get("fullscreened") && (Info.isMobile() && Info.isiOS())) {
+                        this.next("LoadVideo");
+                        if (this.dyn.get("adsplayer_active")) this.dyn.set("adsplayer_active", false)
```

Comment:
Hi @yoryi, going to next state destroys the current one, so calling `this.dyn.get(...)` after `this.next()` throws undefined error.
```suggestion
                        if (this.dyn.get("adsplayer_active")) this.dyn.set("adsplayer_active", false)
                        this.next("LoadVideo");
```